src/Voron/Util/ThreadHoppingReaderWriterLock.cs

```diff
@@ -36,7 +32,7 @@ public void EnterWriteLock()
             var currentWaiters = Interlocked.Add(ref _waiters, WriterMarker);
 
             // try take ownership on lock
-            var currentLock = Interlocked.CompareExchange(ref _writeLockOwnerThreadId, _localThreadIdCache.Value, 0);
+            var currentLock = Interlocked.CompareExchange(ref _writeLockOwnerThreadId, Thread.CurrentThread.ManagedThreadId, 0);
```

Comment:
Wasn't the cache crated because accessing 'Thread.CurrentThread.ManagedThreadId' is expensive? Maybe it is better to access 'NativeMemory.CurrentThreadStats.ManagedThreadId' instead?

Anyway, at least we can create a variable in this method to avoid accessing it twice.

Reply:
It used to be expensive, but now, I don't think so.
See:

https://github.com/dotnet/runtime/blob/master/src/libraries/System.Private.CoreLib/src/System/Threading/Thread.cs#L19

https://github.com/dotnet/runtime/blob/master/src/libraries/System.Private.CoreLib/src/System/Threading/Thread.cs#L156

https://github.com/dotnet/runtime/blob/master/src/coreclr/src/System.Private.CoreLib/src/System/Threading/Thread.CoreCLR.cs#L161

https://github.com/dotnet/runtime/blob/fcd862e06413a000f9cafa9d2f359226c60b9b42/src/coreclr/src/vm/object.h#L1426

So this ends up being as fast as our own impl